index.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+const Executor = require('screwdriver-executor-base');
+const path = require('path');
+const jenkins = require('jenkins');
+const fs = require('fs');
+const async = require('async');
+const Breaker = require('circuit-fuses');
+
+class J5sExecutor extends Executor {
+
+    /**
+     * JenkinsClient command to run
+     * @method jenkinsCommand
+     * @param  {Object}   options            An object that tells what command & params to run
+     * @param  {String}   options.module     Jenkins client module. For example: job, build
+     * @param  {String}   options.action     Jenkins client action in the given module. For example: get, create
+     * @param  {Object}   options.params     Parameters to run with
+     * @param  {Function} callback           fn(err) from jenkinsClient
+     */
+    jenkinsCommand(options, callback) {
+        // To pass arguments as an array, we need to use apply
+        options.params.push(callback);
+
+        return this.jenkinsClient[options.module][options.action].apply(
+            this.jenkinsClient[options.module],
+            options.params);
+    }
+
+    /**
+     * Constructor
+     * @method constructor
+     * @param  {Object} options           Configuration options
+     * @param  {Object} options.host      Jenkins hostname to make requests to
+     * @param  {Object} options.username  Jenkins username
+     * @param  {Object} options.password  Jenkins password/token
+     */
+    constructor(options) {
+        super();
+        this.host = options.host;
+        this.username = options.username;
+        this.password = options.password;
+        // need to pass port nubmer in the future
+        this.baseUrl = `http://${this.username}:${this.password}@${this.host}:8080`;
+        this.jenkinsClient = jenkins({
+            baseUrl: this.baseUrl,
+            crumbIssuer: true
+        });
+
+        this.breaker = new Breaker(this.jenkinsCommand.bind(this));
```

Comment:
We tried that before and that will switch the this context we need for jenkinsClient into Breaker